packages/compass-global-writes/src/store/reducer.ts

```diff
@@ -220,6 +199,9 @@ export type RootState = {
   namespace: string;
   managedNamespace?: ManagedNamespace;
   shardZones: ShardZoneData[];
+  isSubmittingForSharding?: boolean;
+  isCancellingSharding?: boolean;
+  isUnmanagingNamespace?: boolean;
```

Comment:
Using boolean flags as states like that when they are not actually representing potentially overlapping states is an antipattern and a bug waiting to happen: this allows application to be in multiple incompatible states at the same time with no easy way for us to guarantee that we're dealing with all of them properly in the reducer. Having three possible values on three variables like that means we have almost a hundred unique state variations that we need to deal with, this doesn't sound right for the business logic we're implementing here

Reply:
Good point, I was just trying to setup stricter types but it's messy. What do you think about replacing them with `userActionInProgress: undefined | 'submitForSharding' | 'cancelSharding' | 'unmanageNamespace'` ? I will also add checks to ensure that action cannot start if another is in progress